tests/SideBySide/AppConfig.cs

```diff
@@ -74,19 +73,8 @@ public static MySqlConnectionStringBuilder CreateSha256ConnectionStringBuilder()
 		// tests can run much slower in CI environments
 		public static int TimeoutDelayFactor { get; } = (Environment.GetEnvironmentVariable("APPVEYOR") == "True" || Environment.GetEnvironmentVariable("TRAVIS") == "true") ? 6 : 1;
 
-		private static string GetCodeRootPath()
-		{
-#if NET46
-			var currentAssembly = Assembly.GetExecutingAssembly();
-#else
-			var currentAssembly = typeof(AppConfig).GetTypeInfo().Assembly;
-#endif
-			var directory = new Uri(currentAssembly.CodeBase).LocalPath;
-			while (!string.Equals(Path.GetFileName(directory), "MySqlConnector", StringComparison.OrdinalIgnoreCase))
-				directory = Path.GetDirectoryName(directory);
-			return directory;
-		}
-
-		private static string ExpandVariables(string value) => value?.Replace("%TESTDATA%", TestDataPath);
+		private static string ExpandVariables(string value, string replacement) => value?.Replace("%TESTDATA%", replacement);
+		private static string ExpandLocalVariables(string value) => ExpandVariables(value, TestDataPath);
+		private static string ExpandRemoteVariables(string value) => ExpandVariables(value, RemoteTestDataPath);
```

Comment:
There should be two because of the way mysql is loading the file. The "local" calls are pulling from the client's machine, and the "server" ones are pulling from the server's machine. Which could be a completely different plaform, so using the same variable doesn't really make sense. As in my case I was testing the "server" files from "/var/lib/mysql-files" and the "local" files" directly from the git checkout.

We don't really need to have %TESTDATA% and %REMOTESTDATA% as they'll never cross paths. I can add it if you want.